romancal/regtest/test_wfi_image_pipeline.py

```diff
@@ -0,0 +1,306 @@
+
+    input_data = "r0000101001001001001_01101_0001_WFI01_uncal.asdf"
+    rtdata.get_data(f"WFI/image/{input_data}")
+    rtdata.input = input_data
+
+    # Test Pipeline
+    output = "r0000101001001001001_01101_0001_WFI01_cal.asdf"
+    rtdata.output = output
+    args = [
+        "roman_elp",
+        rtdata.input,
+    ]
+    ExposurePipeline.from_cmdline(args)
+
+    # get truth file
+    rtdata.get_truth(f"truth/WFI/image/{output}")
+    return rtdata
+
+
+@pytest.fixture(scope="module")
+def output_filename(run_elp):
+    return run_elp.output
+
+
+@pytest.fixture(scope="module")
+def output_model(output_filename):
+    with rdm.open(output_filename) as model:
+        yield model
+
+
+@pytest.fixture(scope="module")
+def truth_filename(run_elp):
+    return run_elp.truth
+
+
+@pytest.fixture(scope="module")
+def repointed_filename_and_delta(output_filename):
+    delta = [10.0, 10.0]
+
+    with rdm.open(output_filename) as model:
+        # open from the file since we're going to modify the model
+        del model.meta["wcs"]
+
+        # Create new pointing for the model
+        # RA & Dec are each shifted + 10 degrees, unless they are near
+        # the upper limit, in which case they are shifted -10 degrees.
+        if model.meta.wcsinfo.ra_ref >= 350.0:
+            delta[0] *= -1.0
+        if model.meta.wcsinfo.dec_ref >= 80.0:
+            delta[1] *= -1.0
+
+        model.meta.wcsinfo.ra_ref += delta[0]
+        model.meta.wcsinfo.dec_ref += delta[1]
+
+        # Create new wcs object for the new pointing
+        model = AssignWcsStep.call(model)
+
+        # save repointed model
+        repointed_filename = output_filename.rsplit(".", 1)[0] + "_repoint.asdf"
+        model.to_asdf(repointed_filename)
+
+    return repointed_filename, delta
+
+
+@pytest.mark.soctests
+def test_output_matches_truth(output_filename, truth_filename, ignore_asdf_paths):
+    diff = compare_asdf(output_filename, truth_filename, **ignore_asdf_paths)
+    assert diff.identical, diff.report()
+
+
+@pytest.mark.soctests
+def test_output_is_image_model(output_model):
+    # DMS280 result is an ImageModel
+    assert isinstance(output_model, rdm.datamodels.ImageModel)
+
+
+@pytest.mark.soctests
+@pytest.mark.parametrize(
+    "step_name",
+    (
+        "assign_wcs",
+        "flat_field",
+        "dark",
+        "dq_init",
+        "jump",
+        "linearity",
+        "ramp_fit",
+        "saturation",
+    ),
+)
+def test_steps_ran(output_model, step_name):
+    # DMS86
+    # also DMS129 for assign_wcs
+    assert getattr(output_model.meta.cal_step, step_name) == "COMPLETE"
+
+
+@pytest.mark.soctests
+def test_jump_in_uneven_ramp(output_model):
+    # DMS361 jump detection detected jumps in uneven ramp
+    uneven = len({len(x) for x in output_model.meta.exposure.read_pattern}) > 1
+    assert uneven & np.any(output_model.dq & pixel.JUMP_DET)
+
+
+@pytest.mark.soctests
+def test_has_a_wcs(output_model):
+    # DMS129  TODO not listed
```

Comment:
DMS129 isn't listed in the requirements. Is it satisfied?

Reply:
Yes, we should mark this as done.  I can't figure out why it's not picked up by John's page, though.

Reply:
Added DMS129 in 4c0ad1d